Maui.RevenueCat.InAppBilling/Platforms/Android/RevenueCatBillingAndroid.cs

```diff
@@ -56,48 +56,47 @@ public partial class RevenueCatBilling : IRevenueCatBilling
         }
 
         try
         {
-            using var offerings = await Purchases.SharedInstance.GetOfferingsAsync(cancellationToken);
-            if (offerings is null)
+            _cachedOfferingPackages = await Purchases.SharedInstance.GetOfferingsAsync(cancellationToken);
+            if (_cachedOfferingPackages is null)
             {
                 return new();
             }
 
-            using var currentOffering = offerings.Current;
-            if (currentOffering is null)
-            {
-                return new();
-            }
-
-            _cachedOfferingPackages = currentOffering.AvailablePackages.ToList();
-
-            return _cachedOfferingPackages.ToOfferDtoList(); ;
+            return _cachedOfferingPackages.ToOfferingDtoList().ToList();
         }
         catch (Exception ex)
         {
-            _logger.LogError(ex, $"{nameof(LoadOfferings)} didn't suceed.");
+            _logger.LogError(ex, $"{nameof(LoadOfferings)} didn't succeed.");
             return new();
         }
     }
-    public async partial Task<PurchaseResult> PurchaseProduct(string offeringIdentifier, CancellationToken cancellationToken)
+    public async partial Task<PurchaseResult> PurchaseProduct(string offeringIdentifier, string packageIdentifier, CancellationToken cancellationToken)
     {
         if (!_isInitialized)
         {
             _logger.LogError($"To call {nameof(PurchaseProduct)} you firstly have to call Initialize method.");
             throw new Exception("RevenueCatBilling wasn't initialized");
         }
 
-        var packageToBuy = _cachedOfferingPackages.FirstOrDefault(p => p.Identifier == offeringIdentifier);
+        var offeringToBuy = _cachedOfferingPackages.GetOffering(offeringIdentifier);
+        if (offeringToBuy is null)
+        {
+            _logger.LogError("No offering with identifier: {offeringIdentifier} found. Make sure you called LoadOfferings before.", offeringIdentifier);
+            throw new Exception($"No offering with identifier: {offeringIdentifier} found. Make sure you called LoadOfferings before.");
+        }
+
+        var packageToBuy = offeringToBuy.AvailablePackages.FirstOrDefault(p => p.Identifier == packageIdentifier);
         if (packageToBuy is null)
         {
-            _logger.LogError("No offering/packege with identifier: {offeringIdentifier} found. Make sure you called LoadOfferings before.", offeringIdentifier);
-            throw new Exception($"No offering/packege with identifier: {offeringIdentifier} found. Make sure you called LoadOfferings before.");
+            _logger.LogError("No package with identifier: {packageIdentifier} found. Make sure you called LoadOfferings before.", packageIdentifier);
+            throw new Exception($"No offering with identifier: {packageIdentifier} found. Make sure you called LoadOfferings before.");
         }
 
         if (_currentActivityContext is null)
```

Comment:
I know that u havent change anything in condition `if (_currentActivityContext is null)`
but it would probably make more sense to put this condition directly after `if (!_isInitialized)` so it can fail faster